packages/react/modules/hooks/use-adapter-subscription/use-adapter-subscription.ts

```diff
@@ -0,0 +1,18 @@
+import { TAdapter } from '@flopflip/types';
+import React from 'react';
+
+function useAdapterSubscription(adapter: TAdapter) {
+  React.useEffect(() => {
+    if (adapter.subscribe) {
+      adapter.subscribe();
+    }
+
+    return () => {
+      if (adapter.unsubscribe) {
+        adapter.unsubscribe();
+      }
+    };
+  }, [adapter]);
```

Comment:
Maybe better to destructure the subscribe/unsubscribe methods and use those in the dependency list.

Reply:
You mean like 9124ee9, right? It won't help much I think as the adapter itself doesn't change. It's a static import. Unlike props which change as a whole when one prop changes.

I am really unsure what the cause for problems here is. Does the implementation roughly look fine to you? The errors come when we e.g. test `yarn test:watch use-feature-toggle`. Only the broadcast package has it as redux takes care of the it otherwise (the setters are action creators).